feature_engine/selection/recursive_feature_addition.py

```diff
@@ -0,0 +1,269 @@
+
+        # dict to collect features and their performance_drift
+        # It is initialized with the performance drift of
+        # the most important feature
+        self.performance_drifts_ = {
+            first_most_important_feature: baseline_model_performance -
+            self.initial_model_performance_
+        }
+
+        # loop over the ordered list of features by feature importance starting
+        # from the second element in the list.
+        # Train the estimator with
+        for feature in list(self.feature_importances_.index)[1:]:
+
+            # Add feature and train new model
+            model_tmp = cross_validate(
+                self.estimator,
+                X[self.selected_features_ + [feature]],
+                y,
+                cv=self.cv,
+                scoring=self.scoring,
+                return_estimator=True,
+            )
+
+            # assign new model performance
+            model_tmp_performance = model_tmp["test_score"].mean()
+
+            # Calculate performance drift
+            performance_drift = model_tmp_performance - baseline_model_performance
+
+            # Save feature and performance drift
+            self.performance_drifts_[feature] = performance_drift
+
+            # If new performance model is
+            if performance_drift > self.threshold:
+
+                # add feature to the list of selected features
+                self.selected_features_.append(feature)
+
+                # run updated baseline model with the new list of features
+                baseline_model = cross_validate(
+                    self.estimator,
+                    X[self.selected_features_],
+                    y,
+                    cv=self.cv,
+                    scoring=self.scoring,
+                    return_estimator=True,
+                )
+
+                # Save updated baseline model performance
+                baseline_model_performance = baseline_model["test_score"].mean()
+
+        self.input_shape_ = X.shape
+
+        return self
+
+    def transform(self, X: pd.DataFrame):
+        """
+        Removes non-selected features. That is, features which did not cause a big
+        estimator performance drop when removed from the dataset.
+
+        Args
+        ----
+
+        X: pandas dataframe of shape = [n_samples, n_features].
+            The input dataframe from which features will be selected.
+
+        Returns
+        -------
+
+        X_transformed: pandas dataframe
+            of shape = [n_samples, n_selected_features]
+            Pandas dataframe with the selected features.
+        """
+
+        # check if fit is performed prior to transform
+        check_is_fitted(self)
+
+        # check if input is a dataframe
+        X = _is_dataframe(X)
+
+        # check if number of columns in test dataset matches to train dataset
+        _check_input_matches_training_df(X, self.input_shape_[1])
+
```

Comment:
should we check na? I think here we should, because we are not training a model, so it won't be detected otherwise.